library/Icinga/Application/Hook/PasswordPolicyHook.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/* Icinga Web 2 | (c) 2025 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Application\Hook;
+
+interface PasswordPolicyHook
```

Comment:
Given PHP has no multiple inheritance, I wanted to preventively avoid unnecessary base classes and instructed you to make an interface.

See how tables have turned: https://github.com/Icinga/icingaweb2/pull/5442#discussion_r2498048910 🙈

Reply:
This comment from Alex doesn't read like an instruction, so here's the explanation:

We use arbitrary strings such as `PasswordPolicy` to register and load hooks, which makes it difficult to actually find hook usages. I would suggest making `PasswordPolicyHook` an abstract class that also provides the methods `all()` and `register()` so that the string `PasswordPolicy` only appears in the hook itself. Please then also adjust the documentation.